### PR TITLE
Fix @Category annotations on tests

### PR DESCRIPTION
**Pull request by frant-hartm**
Using `@Category` on a subclass doesn't replace the categories on superclass, but it adds to the categories. So if you declare QuickTest and ParallelJVMTest on the parent and only QuickTest on the child it will run in parallel phase in PR builder. Fixed all cases like this in `hazelcast` module.

If the child uses either SlowTest or NightlyTest it is ok, because it will run during respective executions and Quick/Parallel are not considered at all, so I left these cases as they are.

This has been highlighted when we updated surefire/failsafe to 3.0.0-M7, previously used version 2.22.2 has a bug and would ignore these tests.

Possibly affected tests: #21318, #21156

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by srknzl** [comment]
> Using @Category on a subclass doesn't replace the categories on superclass, but it adds to the categories. So if you declare QuickTest and ParallelJVMTest on the parent and only QuickTest on the child it will run in parallel phase in PR builder.

What about runners?  (RunWith annotations)

File: hazelcast/src/test/java/com/hazelcast/cache/CacheListenerTest.java [hunk 1]
```diff
@@ -51,7 +50,7 @@
 import static org.junit.Assert.fail;
 
 @RunWith(HazelcastSerialClassRunner.class)
-@Category({QuickTest.class, ParallelJVMTest.class})
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheListenerTest.java:54]: Why did you remove this? The test does not extend another test class.

File: hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java [hunk 2]
```diff
@@ -40,18 +39,18 @@
...
 import static com.hazelcast.cache.HazelcastCachingProvider.propertiesByLocation;
-import static com.hazelcast.cache.CacheTestSupport.createServerCachingProvider;
 import static com.hazelcast.cache.jsr.JsrTestUtil.cleanup;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 @RunWith(HazelcastSerialClassRunner.class)
-@Category({QuickTest.class, ParallelJVMTest.class})
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java:54]: Similar comment for here as well.

File: hazelcast/src/test/java/com/hazelcast/config/replacer/AbstractPbeReplacerTest.java [hunk 3]
```diff
@@ -40,7 +39,7 @@
  * Unit tests for {@link AbstractPbeReplacer}.
  */
 @RunWith(HazelcastParallelClassRunner.class)
-@Category({ QuickTest.class, ParallelJVMTest.class })
+@Category(QuickTest.class)
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/config/replacer/AbstractPbeReplacerTest.java:42]: I think you changed this because parallel runner and parallel jvm tests should not be set at the same time. Remember it from a slack discussion

File: hazelcast/src/test/java/com/hazelcast/config/replacer/EncryptionReplacerTest.java [hunk 4]
```diff
@@ -49,7 +47,6 @@
  * Unit tests for {@link EncryptionReplacer}.
  */
 @RunWith(HazelcastSerialClassRunner.class)
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/config/replacer/EncryptionReplacerTest.java:49]: Does this class override the runner to be serial?

File: hazelcast/src/test/java/com/hazelcast/internal/ascii/MemcachedMultiendpointTest.java [hunk 5]
```diff
@@ -20,15 +20,8 @@
...
-import com.hazelcast.test.HazelcastParallelClassRunner;
-import com.hazelcast.test.annotation.QuickTest;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
 
-@RunWith(HazelcastParallelClassRunner.class)
-@Category(QuickTest.class)
-public class MemcachedMultiendpointTest
-        extends MemcachedTest {
+public class MemcachedMultiendpointTest extends MemcachedTest {
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/internal/ascii/MemcachedMultiendpointTest.java:24]: MemcachedTest is 

```
@RunWith(HazelcastParallelClassRunner.class)
@Category({QuickTest.class, ParallelJVMTest.class})
```

fyi if you want to remove parallel jvm test from this one as well.

File: hazelcast/src/test/java/com/hazelcast/cache/CacheListenerTest.java (hunk 1 above)
Comment (frant-hartm) [reply hazelcast/src/test/java/com/hazelcast/cache/CacheListenerTest.java:54]: But this class is extended - by `CacheClientListenerTest`, and that class can't be `ParallelJVMTest` because it uses real client.

File: hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java (hunk 2 above)
Comment (frant-hartm) [reply hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java:54]: Same reason as for `CacheListenerTest`

File: hazelcast/src/test/java/com/hazelcast/config/replacer/EncryptionReplacerTest.java (hunk 4 above)
Comment (frant-hartm) [reply hazelcast/src/test/java/com/hazelcast/config/replacer/EncryptionReplacerTest.java:49]: For some reason it does, not sure why.

File: hazelcast/src/test/java/com/hazelcast/internal/ascii/MemcachedMultiendpointTest.java (hunk 5 above)
Comment (frant-hartm) [reply hazelcast/src/test/java/com/hazelcast/internal/ascii/MemcachedMultiendpointTest.java:24]: Yes, I think this one is safe to be ParallelJVMTest, the parent is and we only use slightly different instance config.

**Comment by frant-hartm** [comment]
Runners get overridden correctly.

**Comment by frant-hartm** [comment]
run-lab-run

**Review by TomaszGaweda — APPROVED** [review]

**Review by srknzl — APPROVED** [review]
